src/Sparrow/LowMemory/LowMemoryNotification.cs

```diff
@@ -119,6 +123,46 @@ public LowMemoryNotification(long lowMemoryThreshold)
             thread.Start();
         }
 
+        public static (long WorkingSet, long TotalUnmanagedAllocations, long ManagedMemory) MemoryStatsInternal()
+        {
+            using (var currentProcess = Process.GetCurrentProcess())
+            {
+                var workingSet =
+                    PlatformDetails.RunningOnPosix == false || PlatformDetails.RunningOnMacOsx
+                        ? currentProcess.WorkingSet64
+                        : MemoryInformation.GetRssMemoryUsage(currentProcess.Id);
+
+                long totalUnmanagedAllocations = 0;
+                foreach (var stats in NativeMemory.ThreadAllocations.Values
```

Comment:
Change this to use a direct call. It is going to be called a lot, let's not do all these allocations